pkg/flowkit/services/transactions.go

```diff
@@ -80,6 +77,56 @@ func (t *Transactions) GetStatus(
 	return tx, result, err
 }
 
+// BuildWithSigners builds a transaction with a set of signers.
+func (t *Transactions) BuildWithSigners(
```

Comment:
not sure if I like this extra method on the API. It starts to make API grow for very specific cases.

Reply:
This part is a bit tricky, because as we are getting a lot of data with assumption from signers ( last signer is payer, first one is proposer , first N ( N authorizer count in code ) is authorizers. It is hard to put them as a method of `transaction`. 

For sure it has to be either a NewTransaction constructor or some builder.

Another option is to break this function as utility functions. But the utility function is a bit complicated too, for example one of them needs to be something like 'GetUniqueAddressesFromAccounts`, but it feels so useless and not enough utility.

Reply:
I'm not sure I like this implicitness of the order. I feel we could in case you want to have multiple roles signers just support usage of the same flags as in build (where multiple roles are allowed) to keep the consistency of flag purpose. So I feel if you want to use multiple signers you have to specify them using the flags like `--proposer alice --authorizer alice --payer bob`. What do you think? And because it's a send command that then replaces the single signer where all the roles are done by one account. I feel that also makes the implementation clearer. We can also just use the build then as it's already used that way.

Reply:
Also @bluesign now that I think more about it I feel the `transactions.Send` function itself should be refactored to support passing proposer, payer, authorizers. By default that will all just be same but still.

Reply:
> Also @bluesign now that I think more about it I feel the `transactions.Send` function itself should be refactored to support passing proposer, payer, authorizers. By default that will all just be same but still.

yeah, I think this would be the best solution. Let me shape it up a bit and we discuss.

Reply:
@sideninja one question here: if` --payer bob` is specified but `proposer` is not, should we assume proposer from payer ? I am thinking how we can move to that state from having single `--signer` with less problems

Reply:
I would keep signer for only one that has all roles as ti was before, but if you want to use different roles then you must specify all.